### DISCUSSION TITLE
Add native sharing with other Mobile Apps

### DISCUSSION THREAD

**Discussion by rom-1** [FreshRSS/FreshRSS#3327] [Ideas]
As there is no official FreshRSS App, a lot of people are using the FreshRSS Web Interface with their Mobile Phone (e.g. Chrome on Android). One of the features they are missing here is native sharing a link/article with other apps. To make this a little bit easier, I suggest to add a new sharing option (/i/?c=configure&a=integration), which enables sharing with "Apps". 

There are two ways this could be archived: 

- Best: Add "Share with Mobile Apps" as a Sharing Target, and use the Web Share API, see here: https://web.dev/web-share/
- Alternative: Add "Copy to Clipboard" as a Sharing Target, see here: https://web.dev/async-clipboard/

or maybe even both options. ;)

**Comment by Frenzie** [comment]
As an aside, see https://github.com/FreshRSS/FreshRSS#apis--native-apps

MDN link about this new functionality: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share

**Comment by aledeg** [comment]
@rom-1 I think your idea of clipboard sharing is a good one. I've implemented in #3330.
Feel free to test it and let me know if this works with a smartphone.

**Reply by rom-1** [reply]
Just tested it on Android, Chrome. Works great! Thanks a lot!

[As Alkarex already suggested](https://github.com/FreshRSS/FreshRSS/pull/3330#issuecomment-753648834), a future improvement would be to "auto-close" the menu once it is clicked on.

**Comment by rom-1** [comment]
@aledeg Thanks! I will give it a try once it is merged into master. 
Any specific reason why you are not going for the Web Share API? While the clipboard solution is definitely an improvement, the WebShareAPI would spare the users 2 additional clicks/taps for every share action (switching to another app, long-press to paste content).

**Reply by aledeg** [reply]
@rom-1 The clipboard share could be used also on desktop and is really easy to implement. That's the reason behind #3330 PR. I might as well try the other but maybe not today.
I think you should open a dedicated issue for that.

**Comment by rom-1** [comment]
https://github.com/FreshRSS/FreshRSS/issues/3332

**Comment by aledeg** [comment]
I think we could delete that discussion. We've covered one of the original request and the other one is logged in the backlog.
